test/test-parse-function.cpp

```diff
@@ -1035,9 +1035,16 @@ TEST_F(test_parse_function, arrow_function_with_invalid_parameters) {
     test_parser p(u8"([(x,)] => {});"_sv, capture_diags);
     auto guard = p.enter_function(function_attributes::generator);
     p.parse_and_visit_statement();
-    EXPECT_THAT(p.errors, ElementsAre(DIAG_TYPE_OFFSETS(
-                              p.code, diag_stray_comma_in_parameter,  //
-                              comma, strlen(u8"([(x"), u8",")));
+    EXPECT_THAT(
+        p.errors,
+        ElementsAre(
+            DIAG_TYPE_OFFSETS(                                                //
+                p.code, diag_unexpected_function_parameter_is_parenthesized,  //
+                left_paren_to_right_paren, strlen(u8"(["), u8"(x,)"),         //
+                DIAG_TYPE_OFFSETS(                                            //
+                              p.code, diag_stray_comma_in_parameter,          //
```

Comment:
Hmm, do we need diag_stray_comma_in_parameter anymore after your change? I think the only way to have a stray comma is to have parentheses.